test/sagemaker_tests/mxnet/training/integration/sagemaker/test_horovod.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
+
+import sagemaker
+from sagemaker.mxnet import MXNet
+
+from ...integration import RESOURCE_PATH
+from ...integration.utils import unique_name_from_base
+
+def test_distributed_training_horovod(sagemaker_session,
+                                      instance_type,
+                                      ecr_image,
+                                      tmpdir,
+                                      framework_version):
+
+    mpi_options = '-verbose -x orte_base_help_aggregate=0'
+    estimator = MXNet(
+        entry_point=os.path.join(RESOURCE_PATH, 'mnist', 'horovod_mnist.py'),
+        role='SageMakerRole',
+        train_instance_type=instance_type,
+        train_instance_count=2,
+        image_name=ecr_image,
+        framework_version=framework_version,
+        py_version='py3',
```

Comment:
Could you make this an input from pytest fixture `py_version`?

Please refer to conftest.py.

Reply:
Just like TF, let's test it specifically for py3 for now
https://github.com/aws/deep-learning-containers/blob/3c9a398a4de9e60b22d823c3868f60fb8eeb45d5/test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_horovod.py#L39

Reply:
That's not exactly what happens. The SM Python SDK sets certain internal variables as though it were using "py3" instead of "py2". The image URI itself is already assigned as one of the input parameters, here. So it will use a py2 image if that is what is given as input, regardless of what the py_version parameter says.

For TF, it is related to "script_mode", but for MXNet it only seems to print a deprecation warning for py2. So it's alright.